File: Top 100 Codes/01-Getting Started/19-Power of a number/Powerofanumber.py
# Prompt the user for the base number and the power
num = int(input("Enter the base number: "))
power = int(input("Enter the power: "))

# Calculate the result using the built-in pow() function
result = pow(num, power)

# Display the result
print(f"{num} raised to the power {power} is: {result}")

# Notes:
# - The pow() function efficiently calculates num^power.
# - It works for both positive and negative values of num and power.

File: Top 100 Codes/01-Getting Started/19-Power of a number/Powerofanumber.cpp
#include <iostream>
#include <cmath> // For pow() function
using namespace std;

int main() {
    double base, exponent;

    // Prompt the user for input
    cout << "Enter the base: ";
    cin >> base;
    cout << "Enter the exponent: ";
    cin >> exponent;

    // Calculate the power using pow function
    double result = pow(base, exponent);

    // Display the result
    cout << base << " raised to the power " << exponent << " is: " << result << endl;

    return 0;
}

// Key Notes:
// 1. This method handles all cases, including when base/exponent are integers/decimals, or positive/negative.
// 2. The pow() function is included in the <cmath> library.
// Time Complexity: O(1) - pow() is a constant-time operation for input size.
// Space Complexity: O(1) - Constant space is used.
